pathology/hovernet/infer_roi.py

```diff
@@ -0,0 +1,217 @@
+
+from monai.apps.pathology.inferers import SlidingWindowHoVerNetInferer
+from monai.apps.pathology.transforms import (
+    GenerateDistanceMapd,
+    GenerateInstanceBorderd,
+    GenerateWatershedMarkersd,
+    GenerateWatershedMaskd,
+    HoVerNetNuclearTypePostProcessingd,
+    Watershedd,
+)
+from monai.data import DataLoader, Dataset, PILReader
+from monai.engines import SupervisedEvaluator
+from monai.networks.nets import HoVerNet
+from monai.transforms import (
+    Activationsd,
+    AsDiscreted,
+    CastToTyped,
+    Compose,
+    EnsureChannelFirstd,
+    FillHoles,
+    FromMetaTensord,
+    GaussianSmooth,
+    LoadImaged,
+    PromoteChildItemsd,
+    SaveImaged,
+    ScaleIntensityRanged,
+)
+from monai.utils import HoVerNetBranch, first
+
+
+def create_output_dir(cfg):
+    if cfg["mode"].lower() == "original":
+        cfg["patch_size"] = 270
+        cfg["out_size"] = 80
+    elif cfg["mode"].lower() == "fast":
+        cfg["patch_size"] = 256
+        cfg["out_size"] = 164
+
+    timestamp = time.strftime("%y%m%d-%H%M%S")
+    run_folder_name = f"{timestamp}_inference_hovernet_ps{cfg['patch_size']}"
+    log_dir = os.path.join(cfg["output"], run_folder_name)
+    print(f"Logs and outputs are saved at '{log_dir}'.")
+    if not os.path.exists(log_dir):
+        os.makedirs(log_dir)
+    return log_dir
+
+
+def run(cfg):
+    # --------------------------------------------------------------------------
+    # Set Directory and Device
+    # --------------------------------------------------------------------------
+    output_dir = create_output_dir(cfg)
+    multi_gpu = True if cfg["use_gpu"] and torch.cuda.device_count() > 1 else False
+    if multi_gpu:
+        dist.init_process_group(backend="nccl", init_method="env://")
+        device = torch.device("cuda:{}".format(dist.get_rank()))
+        torch.cuda.set_device(device)
+    else:
+        device = torch.device("cuda" if cfg["use_gpu"] else "cpu")
+
+    # --------------------------------------------------------------------------
+    # Transforms
+    # --------------------------------------------------------------------------
+    # Preprocessing transforms
+    pre_transforms = Compose(
+        [
+            LoadImaged(keys=["image"], reader=PILReader, converter=lambda x: x.convert("RGB")),
+            EnsureChannelFirstd(keys=["image"]),
+            CastToTyped(keys=["image"], dtype=torch.float32),
+            ScaleIntensityRanged(keys=["image"], a_min=0.0, a_max=255.0, b_min=0.0, b_max=1.0, clip=True),
+        ]
+    )
+    # Postprocessing transforms
+    post_transforms = Compose(
+        [
+            PromoteChildItemsd(
+                keys="pred",
+                child_keys=[HoVerNetBranch.NC.value, HoVerNetBranch.NP.value, HoVerNetBranch.HV.value],
+                delete_keys=True,
+            ),
+            Activationsd(keys=HoVerNetBranch.NC.value, softmax=True),
+            AsDiscreted(keys=HoVerNetBranch.NC.value, argmax=True),
```

Comment:
Seems don't need to add L89-L90, since it has been implemented here.
https://github.com/Project-MONAI/MONAI/blob/72130602340847e15f2042b2bb1b64f6f46eeb93/monai/apps/pathology/transforms/post/array.py#L660-L661
Thanks!